trie/proof.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import absolute_import
+
+from trie import Trie
+
+from trie.exceptions import BadTrieProof
+
+
+# TODO: This function needs to be audited to ensure the approach used here to verify a trie proof
+# is correct.
+def verify_proof(root_hash, key, proof):
+    """Verify that the given proof contains a value for key on a trie with the given root hash.
+
+    Returns the value for the key when it is present, an empty bytes when the key is not present,
```

Comment:
Not a big deal but typically when I see a `verify` function, I expect it to return a boolean based on whether or not the verification was successful. I'm not sure how important it is for you to recover the actual value so I'll leave that up to you. It might be overkill but if you need both you could have a`Proof` object that provides each of these separately, which looks like the `Optional` type in other languages.